packages/api/tests/endpoints/test-s3credentials.js

```diff
@@ -6,70 +6,86 @@ const request = require('supertest');
-const assertions = require('../../lib/assertions');
+
+const EarthdataLoginClient = require('../../lib/EarthdataLogin');
+
 const models = require('../../models');
-const {
-  createFakeJwtAuthToken,
-  fakeAccessTokenFactory
-} = require('../../lib/testUtils');
-const {
-  createJwtToken
-} = require('../../lib/token');
+const { fakeAccessTokenFactory } = require('../../lib/testUtils');
 
-process.env.TOKEN_SECRET = randomString();
-let accessTokenModel;
-let userModel;
-let jwtAuthToken;
+process.env.EARTHDATA_CLIENT_ID = randomId('edlID');
+process.env.EARTHDATA_CLIENT_PASSWORD = randomId('edlPW');
+process.env.DISTRIBUTION_REDIRECT_ENDPOINT = 'http://example.com';
+process.env.DISTRIBUTION_ENDPOINT = `https://${randomId('host')}/${randomId('path')}`;
+process.env.AccessTokensTable = randomId('tokenTable');
 
 
+process.env.TOKEN_SECRET = randomId('tokenSecret');
+let accessTokenModel;
+let authorizationUrl;
+
 // import the express app after setting the env variables
-const { app } = require('../../app');
+const { distributionApp } = require('../../app/distribution');
 
-test.before(async () => {
-  process.env.UsersTable = randomString();
-  userModel = new models.User();
-  await userModel.createTable();
 
-  process.env.AccessTokensTable = randomString();
-  accessTokenModel = new models.AccessToken();
+test.before(async () => {
+  accessTokenModel = new models.AccessToken('token');
   await accessTokenModel.createTable();
 
-  jwtAuthToken = await createFakeJwtAuthToken({ accessTokenModel, userModel });
+
+  const getAccessTokenResponse = {
+    accessToken: randomId('accessToken'),
+    refreshToken: randomId('refreshToken'),
+    username: randomId('username'),
+    expirationTime: Date.now() + (60 * 60 * 1000)
+  };
+  authorizationUrl = randomId('authURL');
+
+  sinon.stub(
+    EarthdataLoginClient.prototype,
+    'getAccessToken'
+  ).callsFake(() => getAccessTokenResponse);
+
+  sinon.stub(
+    EarthdataLoginClient.prototype,
+    'getAuthorizationUrl'
+  ).callsFake(() => authorizationUrl);
 });
 
 test.after.always(async () => {
-  await userModel.deleteTable();
   await accessTokenModel.deleteTable();
+  sinon.reset();
 });
 
-test('GET invokes request for credentials with username from authToken', async (t) => {
+test('An authorized s3credential requeste invokes NGAPs request for credentials with username from accessToken cookie', async (t) => {
+  const username = randomId('username');
   const lambdaInstance = lambda();
   const fakeCredential = { Payload: JSON.stringify({ fake: 'credential' }) };
   const invokeFake = sinon.fake.returns({ promise: () => Promise.resolve(fakeCredential) });
   const previousInvoke = lambdaInstance.invoke;
   lambdaInstance.invoke = invokeFake;
 
-  const parsedToken = verifyJwtToken(jwtAuthToken);
+  const accessTokenRecord = fakeAccessTokenFactory({ username });
+  await accessTokenModel.create(accessTokenRecord);
+
   const FunctionName = 'gsfc-ngap-sh-s3-sts-get-keys';
```

Comment:
The net effect of this is what we're not really hitting the NGAP lambda, right? If so, I kind feel like we should change the `FuncitonName` to `fake-lambda` or something to indicate clearly that this is all mocked

Reply:
d1e65a301